organizations/v0/views.py

```diff
@@ -3,21 +3,40 @@
 Views for organizations end points.
 """
 from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import mixins
+from rest_framework import status
 from rest_framework import viewsets
 from rest_framework.authentication import SessionAuthentication
 from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
 from rest_framework_oauth.authentication import OAuth2Authentication
 
 from organizations.models import Organization
 from organizations.serializers import OrganizationSerializer
 
 
-class OrganizationsViewSet(viewsets.ReadOnlyModelViewSet):
-    """Organization view to fetch list organization data or single organization
-    using organization short name.
+class OrganizationsViewSet(mixins.UpdateModelMixin, viewsets.ReadOnlyModelViewSet):
+    """
+    Organization view to:
+        - fetch list organization data or single organization using organization short name.
+        - create or update an organization via the PUT endpoint.
     """
     queryset = Organization.objects.filter(active=True)  # pylint: disable=no-member
     serializer_class = OrganizationSerializer
     lookup_field = 'short_name'
     authentication_classes = (OAuth2Authentication, JwtAuthentication, SessionAuthentication)
     permission_classes = (IsAuthenticated,)
+
+    def update(self, request, *args, **kwargs):
+        short_name = kwargs.get('short_name')
+        organization = Organization.objects.filter(short_name=short_name).first()
+        if organization:
+            return super(OrganizationsViewSet, self).update(request, short_name)
+        serializer = OrganizationSerializer(data=request.data)
+        if serializer.is_valid():
```

Comment:
You probably want to pass `raise_exception=True` here? Which would let you avoid the `if` stanza anyway.